### PR TITLE
Fix passwordHash leak

### PR DESCRIPTION
**Pull request by pbn4**
Fixes https://github.com/bloom-housing/bloom/issues/654

### CODE REVIEW

File: backend/core/src/auth/user.dto.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { OmitType } from "@nestjs/swagger"
+import { User } from "../entity/user.entity"
+import { Expose } from "class-transformer"
+
+export class RegisterResponseDto extends OmitType(User, ["passwordHash", "applications"] as const) {
+  @Expose()
+  accesToken: string
```
Comment (bencpeters) [inline backend/core/src/auth/user.dto.ts:7]: spelling - should be `accessToken`

**Review by bencpeters — APPROVED** [review]
LGTM, minus a spelling change - we should see if that spelling error is just in the DTO, or if it also exists in the returned JSON?

**Review by bk3c — CHANGES_REQUESTED** [review]
Looks ok from a code perspective, but when I run this I get the following error:

`Error: A circular dependency has been detected (property key: "user"). Please, make sure that each side of a bidirectional relationships are using lazy resolvers ("type: () => ClassType").`

File: backend/core/src/auth/user.dto.ts (hunk 1 above)
Comment (bk3c) [reply backend/core/src/auth/user.dto.ts:7]: I agree we should fix this before merge.

Comment (pbn4) [reply backend/core/src/auth/user.dto.ts]: Fixed.

**Comment by pbn4** [comment]
@bk3c  I don't see such warning. Isn't it maybe related to some unstaged code that you have locally?

**Review by bk3c — APPROVED** [review]
Previous errors were resolved after merging in the latest master.

**Pull request merged**